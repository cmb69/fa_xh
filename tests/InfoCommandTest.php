<?php

namespace Fa;

use ApprovalTests\Approvals;
use PHPUnit\Framework\MockObject\MockObject;
use PHPUnit\Framework\TestCase;

class InfoCommandTest extends TestCase
{
    private View $view;

    protected function setUp(): void
    {
        $this->view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["fa"]);
    }

    /** @return InfoCommand&MockObject */
    private function sut()
    {
        return $this->getMockBuilder(InfoCommand::class)
            ->setConstructorArgs(["./plugins/fa/", $this->view])
            ->onlyMethods(["compareVersions", "isWritable"])
            ->getMock();
    }

    public function testRendersPluginInfo(): void
    {
        global $title;
        $response = $this->sut()();
        $this->assertSame("Fa 1.4", $title);
        Approvals::verifyHtml($response);
    }
}
