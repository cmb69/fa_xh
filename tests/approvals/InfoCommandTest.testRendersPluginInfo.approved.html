<!-- info -->
<h1>Fa</h1>
<img src="./plugins/fa/fa.png" class="fa_logo" alt="Fort Awesome">
<p>Version: 1.3</p>
<p>
    Copyright 2017-2021 <a href="http://3-magi.net/" target="_blank">Christoph M.
    Becker</a>
</p>
<p>
    Powered by <a href="http://fontawesome.io" target="_blank">Font Awesome by
    Dave Gandy</a>.
</p>
<div style="clear:both"></div>
<div class="fa_syscheck">
    <h2>System check</h2>
    <p class="xh_fail">Checking that PHP version ≥ 7.1.0 … failure</p>
    <p class="xh_fail">Checking that CMSimple_XH version ≥ 1.7.0 … failure</p>
    <p class="xh_warning">Checking that “./plugins/fa/css/” is writable … warning</p>
    <p class="xh_warning">Checking that “./plugins/fa/config/” is writable … warning</p>
    <p class="xh_warning">Checking that “./plugins/fa/languages/” is writable … warning</p>
</div>
